addons/popochiu/engine/objects/region/popochiu_region.gd

```diff
@@ -33,12 +32,35 @@ func _on_character_entered(chr: PopochiuCharacter) -> void:
 func _on_character_exited(chr: PopochiuCharacter) -> void:
 	pass
 
+func _update_scaling_region(chr: PopochiuCharacter) -> void:
+	var polygon_y_array = []
+	for x in get_node("InteractionPolygon").get_polygon():
+		polygon_y_array.append(x.y)
+
+	chr.on_scaling_region= {
+		'region_description': self.description,
+		'scale_top': self.scale_top, 
+		'scale_bottom': self.scale_bottom,  
+		'polygon_top_y': polygon_y_array.min()+self.position.y+get_node("InteractionPolygon").position.y if self.position else '',
+		'polygon_bottom_y': polygon_y_array.max()+self.position.y+get_node("InteractionPolygon").position.y if self.position else '',
```

Comment:
To prevent lines of code exceeding column 100, I suggest performing the operations to calculate `polygon_top_y` and `polygon_bottom_y` like this.

```suggestion
		'polygon_top_y': (
			polygon_y_array.min()+self.position.y+get_node("InteractionPolygon").position.y
		) if self.position else '',
		'polygon_bottom_y': (
			polygon_y_array.max()+self.position.y+get_node("InteractionPolygon").position.y
		) if self.position else '',
```